src/common/abi/abi.service.ts

```diff
@@ -0,0 +1,68 @@
+import { AbiRegistry, Address, Interaction, IProvider, NetworkConfig, ProxyProvider, QueryResponseBundle, SmartContractAbi } from "@elrondnetwork/erdjs/out";
+import { Logger } from "@nestjs/common";
+import { SmartContractProfiler } from "src/utils/interactions/smartcontract.profiler";
+
+export class AbiService {
+  private readonly proxy: IProvider;
+  private readonly logger: Logger;
+  private readonly abiPath: string;
+  private readonly contractInterface: string;
+  private contract: SmartContractProfiler | undefined;
+
+  constructor(
+    apiUrl: string,
+    abiPath: string,
+    contractInterface: string,
+  ) {
+    this.logger = new Logger(AbiService.name);
+
+    this.proxy = new ProxyProvider(apiUrl);
+
+    this.abiPath = abiPath;
+    this.contractInterface = contractInterface;
+  }
+  public getProxy(): IProvider {
+    return this.proxy;
+  }
+
+  async getContract(contractAddress: string): Promise<SmartContractProfiler> {
+    if (!this.contract) {
```

Comment:
i think it's much more readable if you first check `this.contract !== undefined` then `return this.contract;`